## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#8ddd788e-b4a1-45e8-a5c9-fc501f2b88ad) |
| mint-zoned.sh | :x: [more...](#f7f62c17-73dd-4855-8948-4300448c7869) |
<a name="8ddd788e-b4a1-45e8-a5c9-fc501f2b88ad"></a>
### 12501-54ee0b6/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:30164
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 26adce5f2683:/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... done in 2 seconds
(2/16) Running aws-sdk-java tests ... done in 1 seconds
(3/16) Running aws-sdk-php tests ... done in 44 seconds
(4/16) Running aws-sdk-ruby tests ... done in 3 seconds
(5/16) Running awscli tests ... done in 1 minutes and 27 seconds
(6/16) Running healthcheck tests ... done in 0 seconds
(7/16) Running mc tests ... done in 33 seconds
(8/16) Running minio-dotnet tests ... done in 49 seconds
(9/16) Running minio-go tests ... done in 2 minutes and 28 seconds
(10/16) Running minio-java tests ... done in 1 minutes and 35 seconds
(11/16) Running minio-js tests ... done in 48 seconds
(12/16) Running minio-py tests ... done in 2 minutes and 24 seconds
(13/16) Running s3cmd tests ... FAILED in 9 seconds
{
  "name": "s3cmd",
  "duration": "2944",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-17804/s3cfg get s3://s3cmd-test-bucket-8710/s3cmd-test-object-13325 s3cmd-test-object-13325.downloaded\nProblem: <class 'ConnectionResetError: [Errno 104] Connection reset by peer\nS3cmd:   2.1.0\npython:   3.6.9 (default, Jan 26 2021, 15:33:00) \n[GCC 8.4.0]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3121, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3030, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 604, in cmd_object_get\n    response = s3.object_get(uri, dst_stream, destination, start_position = start_position, extra_label = seq_label)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 726, in object_get\n    response = self.recv_file(request, stream, labels, start_position)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 1733, in recv_file\n    data = http_response.read(this_chunk)\n  File \"/usr/lib/python3.6/http/client.py\", line 463, in read\n    n = self.readinto(b)\n  File \"/usr/lib/python3.6/http/client.py\", line 507, in readinto\n    n = self.fp.readinto(b)\n  File \"/usr/lib/python3.6/socket.py\", line 586, in readinto\n    return self._sock.recv_into(b)\nConnectionResetError: [Errno 104] Connection reset by peer\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}
(13/16) Running s3select tests ... done in 5 seconds
(14/16) Running security tests ... done in 0 seconds
(15/16) Running versioning tests ... done in 3 minutes and 7 seconds

Executed 15 out of 16 tests successfully.
```
<a name="f7f62c17-73dd-4855-8948-4300448c7869"></a>
### 12501-54ee0b6/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:30869
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 1e595d8351bd:/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... done in 1 seconds
(2/16) Running aws-sdk-java tests ... done in 2 seconds
(3/16) Running aws-sdk-php tests ... done in 43 seconds
(4/16) Running aws-sdk-ruby tests ... done in 4 seconds
(5/16) Running awscli tests ... done in 2 minutes and 16 seconds
(6/16) Running healthcheck tests ... done in 0 seconds
(7/16) Running mc tests ... done in 1 minutes and 5 seconds
(8/16) Running minio-dotnet tests ... done in 45 seconds
(9/16) Running minio-go tests ... done in 2 minutes and 3 seconds
(10/16) Running minio-java tests ... done in 1 minutes and 47 seconds
(11/16) Running minio-js tests ... done in 48 seconds
(12/16) Running minio-py tests ... done in 2 minutes and 43 seconds
(13/16) Running s3cmd tests ... done in 20 seconds
(14/16) Running s3select tests ... done in 5 seconds
(15/16) Running security tests ... done in 0 seconds
(16/16) Running versioning tests ... FAILED in 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-v0ug2x60vfrwna",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 616,
  "error": "RequestError: send request failed\ncaused by: Get \"http://minio-dev6.minio.io:30869/versioning-test-v0ug2x60vfrwna/testObject?versionId=03954958-2fee-40bd-8de1-423033c86200\": EOF",
  "function": "testGetObject",
  "message": "GetObject(1) unexpected error with delete marker",
  "name": "versioning",
  "status": "FAIL"
}

Executed 15 out of 16 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/erasure-server-pool.go
+++ b/cmd/erasure-server-pool.go
@@ -661,9 +661,6 @@ func (z *erasureServerPools) GetObjectNInfo(ctx context.Context, bucket, object 
 	for i, res := range results {
 		if res.err == nil {
 			// Select earliest result
-			if res.gr == nil {
-				panic(err)
-			}
 			found = i
 			break
 		}

--- a/cmd/erasure-server-pool.go
+++ b/cmd/erasure-server-pool.go
@@ -666,9 +666,7 @@ func (z *erasureServerPools) GetObjectNInfo(ctx context.Context, bucket, object 
 		}
 		if !isErrObjectNotFound(res.err) && !isErrVersionNotFound(res.err) {
 			for _, res := range results {
-				if res.gr != nil {
-					res.gr.Close()
-				}
+				res.gr.Close()
 			}
 			return nil, res.err
 		}

--- a/cmd/erasure-server-pool.go
+++ b/cmd/erasure-server-pool.go
@@ -670,7 +670,7 @@ func (z *erasureServerPools) GetObjectNInfo(ctx context.Context, bucket, object 
 					res.gr.Close()
 				}
 			}
-			return nil, err
+			return nil, res.err
 		}
 	}
 

